packages/openstef-models/tests/unit/transforms/general/test_imputer.py

```diff
@@ -34,23 +41,88 @@ def sample_dataset() -> TimeSeriesDataset:
     return TimeSeriesDataset(data, timedelta(hours=1))
 
 
+@pytest.fixture
+def correlated_dataset() -> TimeSeriesDataset:
+    """Create sample dataset with simple linear relation between features where temperature = 2 * radiation.
+
+    Returns:
+        A TimeSeriesDataset with 2 features: radiation and temperature (with some NaN),
+        spanning 5 hours.
+    """
+    data = pd.DataFrame(
+        {
+            "radiation": [10.0, np.nan, 30.0, 40.0, 50.0],
+            "temperature": [20.0, 40.0, 60.0, np.nan, 100.0],  # Missing: should be ~80
```

Comment:
Let's remove this comment as I think it is not needed (and tells something about what will be done with the data), and radiation also does not have it
```suggestion
            "temperature": [20.0, 40.0, 60.0, np.nan, 100.0],
```